OpenRA.Game/Network/OrderManager.cs

```diff
@@ -45,6 +45,9 @@ public sealed class OrderManager : IDisposable
 		readonly List<Order> localOrders = new List<Order>();
 		readonly List<Order> localImmediateOrders = new List<Order>();
 
+		readonly List<ClientOrder> processClientOrders = new List<ClientOrder>();
+		readonly List<int> processClientsToRemove = new List<int>();
```

Comment:
No benefit to using queues here - these are just temporary collections that are added to, used, and then cleared within the same method call (to avoid modifying collections while they are being enumerated). I pulled them out here to avoid allocating a new collection each call.

Reply:
On closer inspection `processClientOrders` is consumed by `syncReport.UpdateSyncReport` - and was broken by my initial changes. I fixed that, but changing to a queue would still require an explicit clear step here so I don't think thats worth it.